### DISCUSSION TITLE
Bluetooth Classic using SerialBT

### DISCUSSION THREAD

**Discussion by JIGR1969** [earlephilhower/arduino-pico#1603] [Ideas]
I am looking at a way of being able to specify the local name for when using the Bluetooth Classic Over Serial example and to also extract/determine the MAC address being added to the local name (it replaces the zero's in the function).

I've tried adding a parameter to the begin function within SerialBT.cpp and SerialBT.h, but then ran into trouble further down the line when compiling the code. I believe it is due to the function override parameters conflicting. I can see that within the PicoBluetoothHID.h a function called start is used and has a local name as an entry parameter and it doesn't use the begin function. 

The reasoning for wanting such functionality, is to :

1. Have a more meaningful broadcast name.
2. Be able to display/write out the MAC address such that individual Pico's can be identified (imagine a room with 10 of these and you are trying to connect with a specific one).

Any pointers (pun intended) on how to achieve this, would be gratefully received (if this functionality could be incorporated in a release, that would be much better - don't want to lose functionality if the library is updated).

**Comment by nWestie** [comment]
Sorry, know this is old, let me know if you figured it out well. It looks like the name is generated here: [SerialBT.cpp](https://github.com/earlephilhower/arduino-pico/blob/c62215663fcdf17d29c64daf1c08dc93f4b4844b/libraries/SerialBT/src/SerialBT.cpp#L67C5-L72C58)?
``` cpp
spp_create_sdp_record(_spp_service_buffer, 0x10001, RFCOMM_SERVER_CHANNEL, "PicoW Serial");
sdp_register_service(_spp_service_buffer);

gap_discoverable_control(1);
gap_ssp_set_io_capability(SSP_IO_CAPABILITY_DISPLAY_YES_NO);
gap_set_local_name("PicoW Serial 00:00:00:00:00:00");
```
So you could try editing the library files? I don't know, I'm still trying to figure this out myself

**Reply by piampt** [reply]
I need something like this also. I want to read the BT MAC addr then use the last four digits as a suffix to a name. No success in developing a solution so far.
